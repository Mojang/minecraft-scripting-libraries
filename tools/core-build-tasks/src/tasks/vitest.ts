import { execSync } from 'child_process';

export function vitestTask() {
    return () => {
        execSync('vitest', { stdio: 'inherit' });
    };
}
