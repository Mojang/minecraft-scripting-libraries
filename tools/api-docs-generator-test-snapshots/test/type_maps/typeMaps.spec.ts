// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { describe, it } from 'vitest';
import { runGeneratorForTest } from '../runGeneratorForTest';

describe('Optionals', () => {
    it('Properly generates documentation for modules which contain type maps', () => {
        runGeneratorForTest({
            testDir: __dirname,
            generators: ['ts', 'msdocs'],
        });
    });
});
