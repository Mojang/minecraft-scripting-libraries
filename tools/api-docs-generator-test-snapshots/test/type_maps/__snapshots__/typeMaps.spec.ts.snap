// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Optionals > Properly generates documentation for modules which contain type maps > # Of Files Generated 1`] = `11`;

exports[`Optionals > Properly generates documentation for modules which contain type maps > msdocs/scriptapi/TOC.yml 1`] = `
"  - name: type-maps-and-script-aliases-module
    items:
    - name: type-maps-and-script-aliases-module
      href: type-maps-and-script-aliases-module/type-maps-and-script-aliases-module.md
  - name: type-maps-module
    items:
    - name: type-maps-module
      href: type-maps-module/type-maps-module.md
    - name: CustomComponent
      href: type-maps-module/CustomComponent.md
    - name: FakeComponent
      href: type-maps-module/FakeComponent.md
    - name: TestComponent
      href: type-maps-module/TestComponent.md
"
`;

exports[`Optionals > Properly generates documentation for modules which contain type maps > msdocs/scriptapi/type-maps-and-script-aliases-module/TestTypeMap.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: type-maps-and-script-aliases-module.TestTypeMap Type Alias
description: Contents of the type-maps-and-script-aliases-module.TestTypeMap type alias.
---
# TestTypeMap Type Alias

## Declaration
\`\`\`ts
type TestTypeMap = {
  'FakeMappingOne': FakeTypeOne;
  'FakeMappingTwo': FakeTypeTwo;
  'test:FakeMappingOne': FakeTypeOne;
  'test:FakeMappingTwo': FakeTypeTwo;
};
\`\`\`
"
`;

exports[`Optionals > Properly generates documentation for modules which contain type maps > msdocs/scriptapi/type-maps-and-script-aliases-module/TestTypeMapUnion.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: type-maps-and-script-aliases-module.TestTypeMapUnion Type Alias
description: Contents of the type-maps-and-script-aliases-module.TestTypeMapUnion type alias.
---
# TestTypeMapUnion Type Alias

## Declaration
\`export type TestTypeMapUnion = keyof TestTypeMap;\`
"
`;

exports[`Optionals > Properly generates documentation for modules which contain type maps > msdocs/scriptapi/type-maps-and-script-aliases-module/type-maps-and-script-aliases-module.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: type-maps-and-script-aliases-module Module
description: Contents of the type-maps-and-script-aliases-module module
---
# \`type-maps-and-script-aliases-module\` Module

## Manifest Details
\`\`\`json
{
    "module_name": "type-maps-and-script-aliases-module",
    "version": "1.0.0"
}
\`\`\`
This is version 1.x.x of this module, which is the latest as of version 1.0.0 of Minecraft.

## Available Versions
- \`1.0.0\`

# Type Aliases
- [TestTypeMap](TestTypeMap.md)
- [TestTypeMapUnion](TestTypeMapUnion.md)
"
`;

exports[`Optionals > Properly generates documentation for modules which contain type maps > msdocs/scriptapi/type-maps-module/CustomComponent.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: type-maps-module.CustomComponent Class
description: Contents of the type-maps-module.CustomComponent class.
---
# CustomComponent Class
"
`;

exports[`Optionals > Properly generates documentation for modules which contain type maps > msdocs/scriptapi/type-maps-module/FakeComponent.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: type-maps-module.FakeComponent Class
description: Contents of the type-maps-module.FakeComponent class.
---
# FakeComponent Class
"
`;

exports[`Optionals > Properly generates documentation for modules which contain type maps > msdocs/scriptapi/type-maps-module/TestComponent.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: type-maps-module.TestComponent Class
description: Contents of the type-maps-module.TestComponent class.
---
# TestComponent Class
"
`;

exports[`Optionals > Properly generates documentation for modules which contain type maps > msdocs/scriptapi/type-maps-module/TestComponentTypeMap.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: type-maps-module.TestComponentTypeMap Type Alias
description: Contents of the type-maps-module.TestComponentTypeMap type alias.
---
# TestComponentTypeMap Type Alias

## Declaration
\`\`\`ts
type TestComponentTypeMap = {
  'custom:customComponent': CustomComponent;
  'customComponent': CustomComponent;
  'fake:fakeComponent': FakeComponent;
  'fakeComponent': FakeComponent;
  'test:testComponent': TestComponent;
  'testComponent': TestComponent;
};
\`\`\`
"
`;

exports[`Optionals > Properly generates documentation for modules which contain type maps > msdocs/scriptapi/type-maps-module/type-maps-module.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: type-maps-module Module
description: Contents of the type-maps-module module
---
# \`type-maps-module\` Module

## Manifest Details
\`\`\`json
{
    "module_name": "type-maps-module",
    "version": "1.0.0"
}
\`\`\`
This is version 1.x.x of this module, which is the latest as of version 1.0.0 of Minecraft.

## Available Versions
- \`1.0.0\`

# Type Aliases
- [TestComponentTypeMap](TestComponentTypeMap.md)

## Classes
- [CustomComponent](CustomComponent.md)
- [FakeComponent](FakeComponent.md)
- [TestComponent](TestComponent.md)
"
`;

exports[`Optionals > Properly generates documentation for modules which contain type maps > typescript/type-maps-and-script-aliases-module@1.0.0.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @packageDocumentation
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "type-maps-and-script-aliases-module",
 *   "version": "1.0.0"
 * }
 * \`\`\`
 *
 */
export type TestTypeMap = {
    FakeMappingOne: FakeTypeOne;
    FakeMappingTwo: FakeTypeTwo;
    'test:FakeMappingOne': FakeTypeOne;
    'test:FakeMappingTwo': FakeTypeTwo;
};

export type TestTypeMapUnion = keyof TestTypeMap;
"
`;

exports[`Optionals > Properly generates documentation for modules which contain type maps > typescript/type-maps-module@1.0.0.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @packageDocumentation
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "type-maps-module",
 *   "version": "1.0.0"
 * }
 * \`\`\`
 *
 */
export type TestComponentTypeMap = {
    'custom:customComponent': CustomComponent;
    customComponent: CustomComponent;
    'fake:fakeComponent': FakeComponent;
    fakeComponent: FakeComponent;
    'test:testComponent': TestComponent;
    testComponent: TestComponent;
};

export class CustomComponent {
    private constructor();
}

export class FakeComponent {
    private constructor();
}

export class TestComponent {
    private constructor();
}
"
`;
